drivers/hmis/app/models/hmis/ce/referral_ce_event_manager.rb

```diff
@@ -0,0 +1,73 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Hmis::Ce
+  class ReferralCeEventManager
+    attr_reader :referral
+
+    def initialize(referral)
+      @referral = referral
+    end
+
+    def create_ce_event(message)
+      enrollment = referral.source_enrollment
+      raise 'Referral does not have a source enrollment' unless enrollment.present?
+
+      target_project = referral.target_project
+
+      enrollment.events.create!(
+        event_date: Date.current,
+        event: project_to_event_type(target_project), # TODO(#7527) determine event type from configuration if present
+        location_crisis_or_ph_housing: target_project.id,
+        user: Hmis::Hud::User.from_user(message.user),
+      )
+    end
+
+    def set_ce_event_result(message) # rubocop:disable Naming/AccessorMethodName
+      enrollment = referral.source_enrollment
+      event_type = project_to_event_type(referral.target_project)
+      event = enrollment.events.where(event: event_type).order(:date_created).last # We don't have a precise way to map to the CE Event that was created earlier in this workflow, so just expect it to be the most recently created one.
```

Comment:
Flagging this as a possible problem. We don't directly create a mapping from the CE Event to the referral that generated it. So when it comes time to update that event, we're guessing. Scenario where this could cause a problem:
- user starts Referral 1 for PSH Project 1. CE Event 1 is created
- user starts Referral 2 for PSH Project 2. CE Event 2 is created
- user finishes Referral 1. CE Event 2 would then be updated with the result from Referral 1, which is incorrect.

Does https://github.com/open-path/Green-River/issues/7835 sufficiently protect us from that scenario, or do we need to guard against it further? e.g., add optional `ce_referral_id` col to `Event` table?

Reply:
Created https://github.com/open-path/Green-River/issues/7954 (same ticket as above, as @ttoomey pointed out they are related)